lib/cloudregister/registerutils.py

```diff
@@ -2700,30 +2699,31 @@ def __set_registries_conf_docker(private_registry_fqdn):
         if failed:
             return False
 
+    modified_by_us = False
+    # Setup registry-mirrors in docker daemon.json
     registry_mirrors = docker_cfg_json.get('registry-mirrors', [])
-    modified = False
-    if registry_mirrors:
-        priv_index = -1
-        pub_index = -1
+    if private_registry_url not in registry_mirrors:
+        # susecloud registry not added, always place it first
+        registry_mirrors.insert(0, private_registry_url)
+        modified_by_us = True
 
-        if private_registry_url in registry_mirrors:
-            priv_index = registry_mirrors.index(private_registry_url)
-        if public_registry_url in registry_mirrors:
-            pub_index = registry_mirrors.index(public_registry_url)
-
-        if not priv_index == 0 or not pub_index == 1:
-            if priv_index > 0:
-                registry_mirrors.pop(priv_index)
-            if pub_index > 0:
-                registry_mirrors.pop(pub_index)
-            modified = True
+    if suse_registry_url not in registry_mirrors:
+        # the suse registry search is provided by the libcontainers-common
+        # package. For the case when we cannot find any suse registry we
+        # append it after the susecloud registry
+        #
+        # Note: docker search is disabled for Docker server side !
+        private_registry_index = registry_mirrors.index(
+            private_registry_url
+        )
+        registry_mirrors.insert(
+            private_registry_index + 1, suse_registry_url
+        )
+        modified_by_us = True
+    docker_cfg_json['registry-mirrors'] = registry_mirrors
```

Comment:
no because `modified_by_us` is then False and nothing happens. Actually it doesn't matter where this line appears and I want to make sure it is set only once with data. You are right if `registry_mirrors` stays unchanged there is an assignment of the same data, however python is smart enough not do anything if the address hasn't changed. From that perspective I value simplicity over adding an assignment conditional, because that can again lead to a wrong result

Reply:
I can put the assignment of everything inside of the `if modified_by_us` block but I thought it's an easier read to see the assignment of data at the place where a potential modification could happen

Reply:
that is fine then but
> From that perspective I value simplicity over adding an assignment conditional

I understood we wanted to also value being explicit over implicit

Reply:
ok I moved it, please take another look. Thanks